File: power2orig.php
<?php

$init = 1000;
$start_time = microtime(true);

while (true) {
	if (strlen($init) % 2 != 0) {
		$init = $init * 10;
	}

	$splitter = strlen($init) / 2;
	$fore = substr($init, 0, $splitter);
	$post = substr($init, $splitter, strlen($init) - 1);

	if (($fore + $post)**2 == $init) {
		printf("%d %d %d %f\n", $fore, $post, $init, microtime(true) - $start_time);
	}

	$init = $init + 1;
}

?>

File: power2orig.pl
#!/usr/bin/env perl

use warnings;
use strict;

my $init = 1000;
my $init_time = time();

while (1) {
	if (length($init) % 2 != 0) {
		$init = $init * 10;
	}
	my $splitter = length($init) / 2;

	my $fore = substr $init, 0, $splitter;
	my $post = substr $init, $splitter, $splitter;

	if (($fore + $post) ** 2 == $init) {
		my $found_time = time() - $init_time;
		print "$fore $post $init $found_time\n";
	}
	$init++;
}
